r-vm.tf

```
@@ -209,3 +209,18 @@ resource "azurerm_windows_virtual_machine" "this" {
     ]
   }
 }
+
+data "azurerm_managed_disk" "this" {
+  name                = local.virtual_machine.os_disk[0].name
+  resource_group_name = var.resource_group_name
+}
+
+resource "azapi_resource_action" "this" {
+  type        = "Microsoft.Compute/disks@2022-03-02"
```

```suggestion
  type        = "Microsoft.Compute/disks@2024-03-02"
```

Please consider using the latest API.